x-pack/platform/plugins/shared/cases/public/components/user_actions/property_actions/event_property_actions.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+
+import { AttachmentActionType } from '../../../client/attachment_framework/types';
+import { useCasesContext } from '../../cases_context/use_cases_context';
+import { DeleteAttachmentConfirmationModal } from '../delete_attachment_confirmation_modal';
+import { UserActionPropertyActions } from './property_actions';
+import * as i18n from './translations';
+import { useDeletePropertyAction } from './use_delete_property_action';
+
+interface Props {
+  isLoading: boolean;
+  totalEvents: number;
+  onDelete: () => void;
+}
+
+const EventPropertyActionsComponent: React.FC<Props> = ({ isLoading, totalEvents, onDelete }) => {
+  const { permissions } = useCasesContext();
+  const { showDeletionModal, onModalOpen, onConfirm, onCancel } = useDeletePropertyAction({
+    onDelete,
+  });
+
+  const propertyActions = useMemo(() => {
+    const showRemoveEventIcon = permissions.delete;
+
+    return [
+      ...(showRemoveEventIcon
+        ? [
+            {
+              type: AttachmentActionType.BUTTON as const,
+              color: 'danger' as const,
+              disabled: false,
+              iconType: 'minusInCircle',
+              label: i18n.REMOVE_EVENTS(totalEvents),
+              onClick: onModalOpen,
+            },
+          ]
+        : []),
+    ];
+  }, [permissions.delete, totalEvents, onModalOpen]);
+
+  return (
+    <>
```

Comment:
Is this fragment still needed?

Reply:
yes, there is a modal component adjacent to `UserActionPropertyActions`